### PR TITLE
Set process label to `process_single` for modules with no `task.cpus` usage

### PR DESCRIPTION
**Pull request by jfy133**
I went through all modules I am using in various pipeline/are familiar to me and replaced the `label` from the original multi-core to the new `process_single` for any module that did not have `task.cpu` anywhere in the module code.

Note I also fixed the DEEPARG version placement as I noticed that that appears to have been missed in the recent mass-update of that.

Also includes the following fixes to tools that had broken tests:

- adapterremovalfixprefix - removed version yml from check
- entrezedirect/summary - use string rather than md5sum due to XML variation
- entrezdirect/xtract - use correct output files (the txt with the results, not the input XML)
- harmonization/summarize - fixed out of date test data specification 
- damageprofiler - removed a lot of md5sums of PDFs and added some md5sums of text files

The following module CI test will likely not pass:

- damageprofiler - will be broken until https://github.com/nf-core/test-datasets/pull/619
- mOTUs - requires large database and 2/3 nodes will run out of space
- deepBGC - database download throttling causing database download to fail 

## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: tests/modules/damageprofiler/test.yml [hunk 1]
```diff
@@ -1,110 +1,77 @@
...
     - path: output/damageprofiler/test/edit_distance.pdf
     - path: output/damageprofiler/test/edit_distance.svg
-    - path: output/damageprofiler/test/editDistance.txt
-      md5sum: 04d14b449a5afa8b5dbff0dfa762356b
-    - path: output/damageprofiler/test/Length_plot_combined_data.svg
-    - path: output/damageprofiler/test/Length_plot_forward_reverse_separated.svg
-    - path: output/damageprofiler/test/Length_plot.pdf
     - path: output/damageprofiler/test/lgdistribution.txt
-      md5sum: df2e19195185ea9ee05e8e84b2948f36
+      md5sum: 583fd98094de3fb462b133daeaadcf52
```
Comment (FriederikeHanssen) [inline tests/modules/damageprofiler/test.yml:67]: why do the md5sums change 😱

**Review by FriederikeHanssen — COMMENTED** [review]
nice

**Review by jasmezz — APPROVED** [review]
"One PR per module" eh? 😛

File: tests/modules/damageprofiler/test.yml (hunk 1 above)
Comment (jfy133) [reply tests/modules/damageprofiler/test.yml:67]: There is slight deterministic behaviour IIRC... that one I want to check though once the chr22 -> chr21 is in :)

**Comment by jfy133** [comment]
> "One PR per module" eh? stuck_out_tongue

one per NEW module :shushing_face:

File: tests/modules/damageprofiler/test.yml (hunk 1 above)
Comment (jfy133) [reply tests/modules/damageprofiler/test.yml:67]: This one I mena: https://github.com/nf-core/test-datasets/pull/619

**Review by muffato — APPROVED** [review]

**Comment by muffato** [comment]
Thank you very much @jfy133 , that's really useful !

**Comment by muffato** [comment]
I changed a few more: software that I use/know and can confirm are single-threaded

**Pull request merged**